sentry_sdk/integrations/django/views.py

```diff
@@ -0,0 +1,53 @@
+import functools
+
+from sentry_sdk.hub import Hub
+from sentry_sdk._types import MYPY
+
+if MYPY:
+    from typing import Any
+
+    from django.urls.resolvers import ResolverMatch
+
+
+def patch_resolver():
+    # type: () -> None
+    try:
+        from django.urls.resolvers import URLResolver
+    except ImportError:
+        from django.core.urlresolvers import RegexURLResolver as URLResolver
+
+    from sentry_sdk.integrations.django import DjangoIntegration
+
+    old_resolve = URLResolver.resolve
+
+    def resolve(self, path):
+        # type: (URLResolver, Any) -> ResolverMatch
+        hub = Hub.current
+        integration = hub.get_integration(DjangoIntegration)
+
+        if integration is None or not integration.middleware_spans:
+            return old_resolve(self, path)
+
+        with hub.start_span(op="django.urls.resolve"):
+            return _wrap_resolver_match(hub, old_resolve(self, path))
+
+    URLResolver.resolve = resolve
+
+
+def _wrap_resolver_match(hub, resolver_match):
+    # type: (Hub, ResolverMatch) -> ResolverMatch
+
+    # XXX: The wrapper function is created for every request. Find more
+    # efficient way to wrap views (or build a cache?)
```

Comment:
Yes, it makes sense to compute the wrapped resolver only once. Looks like given a `path` we always return the same `ResolverMatch`.

Reply:
resolvermatch is instantiated on every request so unfortunately we can't use it as cachekey. We could use `resolver_match.func`, which has stable IDs afaict

Reply:
Disregard, the `ResolverMatch` also carries request-specific args and kwargs.
If we cache it would be in the `func` level.